lib/coordinates.rb

```diff
@@ -0,0 +1,59 @@
+class Coordinates
+  def initialize(coords)
+    @coords = coords.split(" ")
+  end
+
+  def are_diagonal?
+    not are_vertical? and not are_horizontal?
+  end
+
+  def are_vertical?
+    expected_length = @coords.length
+    y_coords = @coords.map do |coord|
+      coord[1..-1].to_i
+    end
+    y_coords.uniq.length == 1
+  end
+
+  def are_horizontal?
+    expected_length = @coords.length
+    x_coords = @coords.map do |coord|
+      coord[0]
+    end
+    x_coords.uniq.length == 1
+  end
+
+  def intersects?(coords)
+    coords_a = coords.to_a
+    for index in (0..@coords.length) do
+      coord = @coords[index]
+      return true if coords_a.include?(coord)
+    end
+    false
```

Comment:
Perhaps be consistent with using `return` statements. See below line 53.

Reply:
Yes I agree